pandas/tests/tools/test_to_datetime.py

```diff
@@ -620,16 +625,20 @@ def test_to_datetime_YYYYMMDD(self):
     def test_to_datetime_unparsable_ignore(self):
         # unparsable
         ser = "Month 1, 1999"
-        assert to_datetime(ser, errors="ignore") == ser
+        with tm.assert_produces_warning(UserWarning, match="Could not infer format"):
+            assert to_datetime(ser, errors="ignore") == ser
 
     @td.skip_if_windows  # `tm.set_timezone` does not work in windows
     def test_to_datetime_now(self):
         # See GH#18666
         with tm.set_timezone("US/Eastern"):
             # GH#18705
             now = Timestamp("now")
-            pdnow = to_datetime("now")
-            pdnow2 = to_datetime(["now"])[0]
+            with tm.assert_produces_warning(
+                UserWarning, match="Could not infer format"
+            ):
+                pdnow = to_datetime("now")
```

Comment:
Have we given any consideration to making an exception for the `"now"` literal? The warning here feels a little strange

Reply:
yeah that's a good point - I've now excluded `'now'` and `'today'` from `guess_datetime_format`, so the format will be guessed from the first non-null string that also isn't `'now'` or `'today'`